sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -0,0 +1,1501 @@
+        timeout: int = constants.DEFAULT_TIMEOUT,
+    ):
+        """List of all MXJobs in namespace.
+
+        Args:
+            namespace: Namespace to list the MXJobs.
+            timeout: Optional, Kubernetes API server timeout in seconds
+                to execute the request.
+
+        Returns:
+            list[KubeflowOrgV1MXJob]: List of MXJobs objects. It returns
+            empty list if MXJobs cannot be found.
+
+        Raises:
+            TimeoutError: Timeout to list MXJobs.
+            RuntimeError: Failed to list MXJobs.
+        """
+
+        return utils.list_jobs(
+            custom_api=self.custom_api,
+            api_client=self.api_client,
+            namespace=namespace,
+            job_model=models.KubeflowOrgV1MXJob,
+            job_kind=constants.MXJOB_KIND,
+            job_plural=constants.MXJOB_PLURAL,
+            timeout=timeout,
+        )
+
+    def delete_mxjob(
+        self,
+        name: str,
+        namespace: str = utils.get_default_target_namespace(),
+        delete_options: client.V1DeleteOptions = None,
+    ):
+        """Delete the MXJob
+
+        Args:
+            name: Name for the MXJob.
+            namespace: Namespace for the MXJob.
+            delete_options: Optional, V1DeleteOptions to set while deleting
+                the MXJob. For example, grace period seconds.
+
+        Raises:
+            TimeoutError: Timeout to delete MXJob.
+            RuntimeError: Failed to delete MXJob.
+        """
+
+        utils.delete_job(
+            custom_api=self.custom_api,
+            name=name,
+            namespace=namespace,
+            job_kind=constants.MXJOB_KIND,
+            job_plural=constants.MXJOB_PLURAL,
+            delete_options=delete_options,
+        )
+
+    # ------------------------------------------------------------------------ #
+    # XGBoostJob Training Client APIs.
+    # ------------------------------------------------------------------------ #
+    def create_xgboostjob(
+        self,
+        xgboostjob: models.KubeflowOrgV1XGBoostJob,
+        namespace=utils.get_default_target_namespace(),
+    ):
+        """Create the XGBoostJob.
+
+        Args:
+            xgboostjob: XGBoostJob object of type KubeflowOrgV1XGBoostJob.
+            namespace: Namespace for the XGBoostJob.
+
+        Raises:
+            TimeoutError: Timeout to create XGBoostJob.
+            RuntimeError: Failed to create XGBoostJob.
+        """
+
+        utils.create_job(
+            custom_api=self.custom_api,
+            job=xgboostjob,
+            namespace=namespace,
+            job_kind=constants.XGBOOSTJOB_KIND,
+            job_plural=constants.XGBOOSTJOB_PLURAL,
+        )
+
+    def create_xgboostjob_from_func(self):
+        """Create XGBoost from the function.
+        TODO (andreyvelich): Implement this function.
+        """
+        logging.warning("This API has not been implemented yet.")
```

Comment:
Would you like to work on this in this PR? Or another PR?

Reply:
Let's do that additional APIs in the following PRs @tenzen-y (maybe in the next release).